### PR TITLE
feat: Progress.Circle support conic

### LINKED ISSUE #19397: Conic-radial gradient for Progress (types 'circle' and 'dashboard')
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently you can only specify linear gradient for `Progress`, and when doing a circular type, such as `circle` or `dashboard` this can result in weird behaviour.

Example: [https://codesandbox.io/s/bold-mcnulty-xqpg2](https://codesandbox.io/s/bold-mcnulty-xqpg2)

### What does the proposed API look like?
If allowing conic-gradient in the `stroke-color` could solve this problem:

[Conic-gradient](https://developer.mozilla.org/en-US/docs/Web/CSS/conic-gradient)

Api could look like this:
```typescript
  <Progress
    type="dashboard"
    percent={75}
    strokeColor={{
      "50deg": "#ff0000",
      "100deg": "#ffaa00",
      "150deg": "#ffff00",
      "360deg": "#00ff00"
    }}
  />
```

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #19397

### 💡 Background and solution

截屏2023-08-24 17 20 59

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Refactor Progress `circle` or `dashboard` linear gradient to conic gradient.         |
| 🇨🇳 Chinese |    调整 Progress `circle` 和 `dashboard` 的线性渐变色为锥形渐变色。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 66246a8</samp>

This pull request adds support for conical gradient for `circle` and `dashboard` types of progress, and updates the documentation and demo code accordingly. It also fixes some lint errors and improves the type definitions in `./components/progress/progress.tsx`.

### <samp>🤖 Generated by Copilot at 66246a8</samp>

*  Add support for conical gradient for `circle` and `dashboard` types of progress ([link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-4cf2a9bfa8b6dd431bdb0eee4ce5aff74377e275d7943dfa9162144a5fb80259L3-R7), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-83c9f102d9fda44f8fa96db1119f234832df4434364d31776825d7525bdf5616L4-R21), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL68-R68), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL69-R69), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R24), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L142-R142))
*  Update the demo code and documentation for gradient line progress to include examples and explanations of conical gradient ([link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-4cf2a9bfa8b6dd431bdb0eee4ce5aff74377e275d7943dfa9162144a5fb80259L3-R7), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-83c9f102d9fda44f8fa96db1119f234832df4434364d31776825d7525bdf5616L4-R21), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL68-R68), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL69-R69))
*  Improve the type inference and validation of `strokeColor` and `status` properties in `./components/progress/progress.tsx` using template literal types and partial record ([link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R24), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL41-R44), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL85-R88))
*  Fix lint errors in `./components/progress/progress.tsx` by adding `React` import and moving `omit` import to a new line ([link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR1), [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL7-R8))

### CODE REVIEW

**Comment by afc163** [comment]
什么是锥形渐变色？

File: components/progress/demo/gradient-line.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
 import React from 'react';
 import { Progress, Space } from 'antd';
 
+const twoColors = { '0%': '#108ee9', '100%': '#87d068' };
+const conicColors = { conic: true, '0%': '#87d068', '50%': '#ffe58f', '100%': '#ffccc7' };
```
Comment (afc163) [inline components/progress/demo/gradient-line.tsx:5]: 原来的渐变方式其实一直是错的（和线性 progress 的渐变方式是不一样的），我建议不加新属性，直接改掉。

**Comment by afc163** [comment]
大了好多啊。。

**Review by MadCcc — APPROVED** [review]

File: components/progress/demo/gradient-line.tsx (hunk 1 above)
Comment (zombieJ) [reply components/progress/demo/gradient-line.tsx:5]: 过去实现是通过转角度模拟的，因为 conic-gradient 不是所有浏览器都支持。如果改变实现方式，会导致旧版浏览器 break。

Comment (zombieJ) [reply components/progress/demo/gradient-line.tsx:5]: 折中一下，先检测是否支持再做自动转换如何

Comment (afc163) [reply components/progress/demo/gradient-line.tsx:5]: 可以的。

**Comment by AngelicElegy** [comment]
> [[中文版模板/中文模板](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]
> 
> ### 🤔 这是一个...
> * [x] 新功能
> * [ ] 错误修复
> * [ ] 网站/文档更新
> * [ ] 演示更新
> * [ ] 组件样式更新
> * [ ] TypeScript 定义更新
> * [ ] 捆绑包大小优化
> * [ ] 性能优化
> * [ ] 增强功能
> * [ ] 国际化
> * [ ] 重构
> * [ ] 代码风格优化
> * [ ] 测试用例
> * [ ] 分支合并
> * [ ] 工作流
> * [ ] 其他（关于什么？）
> 
> ### 🔗 相关问题链接
> 解决 #19397
> 
> ### 💡 背景和解决方案
> 截屏2023-08-24 17 20 59
> ### 📝 更新日志
> 语言	变更日志
> 🇺🇸 英语	将进度`circle`或`dashboard`线性渐变重构为圆锥渐变。
> 🇨🇳 中文	调整Progress`circle`和`dashboard`的线性渐变颜色为渐变渐变颜色。
> ### ☑️ 合并前自检
> ⚠️在请求审核之前，请检查以下所有项目。⚠️
> 
> * [x] 文档已更新/提供或不需要
> * [x] 演示已更新/提供或不需要
> * [x] TypeScript 定义已更新/提供或不再需要
> * [x] 提供或不需要变更日志
> 
> ### 🚀 摘要
> ### 🤖 由 Copilot 于[66246a8生成](https://github.com/ant-design/ant-design/commit/66246a81958ee74b80b903841916acd19b6e1cde)
> 此拉取请求添加了对圆锥渐变`circle`和`dashboard`进度类型的支持，并相应地更新了文档和演示代码。它还修复了一些 lint 错误并改进了 中的类型定义`./components/progress/progress.tsx`。
> 
> ### 🔍 演练
> ### 🤖 由 Copilot 于[66246a8生成](https://github.com/ant-design/ant-design/commit/66246a81958ee74b80b903841916acd19b6e1cde)
> * 添加对圆锥渐变`circle`和`dashboard`进度类型的支持（[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-4cf2a9bfa8b6dd431bdb0eee4ce5aff74377e275d7943dfa9162144a5fb80259L3-R7)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-83c9f102d9fda44f8fa96db1119f234832df4434364d31776825d7525bdf5616L4-R21)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL68-R68)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL69-R69)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R24)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L142-R142)）
> * 更新渐变线进度的演示代码和文档，以包含圆锥渐变的示例和解释（[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-4cf2a9bfa8b6dd431bdb0eee4ce5aff74377e275d7943dfa9162144a5fb80259L3-R7)，[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-83c9f102d9fda44f8fa96db1119f234832df4434364d31776825d7525bdf5616L4-R21)，[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL68-R68)，[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL69-R69)）
> * 改进使用模板文字类型和部分记录（[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R24)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL41-R44)、[链接](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL85-R88)）的类型推断和`strokeColor`属性验证`status``./components/progress/progress.tsx`[](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL18-R24)[](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL41-R44)[](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL85-R88)
> * `./components/progress/progress.tsx`通过添加`React`import 并将`omit`import 移至新行来修复 lint 错误( [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR1) , [link](https://github.com/ant-design/ant-design/pull/44404/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL7-R8) )

这个渐变色有bug，conic的deg存在问题，会导致有仪表图渐变色被截断
截屏2025-01-16 15 53 15

**Pull request merged**